cmake/iotjs.cmake

```diff
@@ -504,16 +504,24 @@ else()
 
   # FIXME: module specific condition should not be in the main cmake
   if(${ENABLE_MODULE_NAPI})
+    # Some tests require the GC to be exposed
+    iotjs_add_compile_flags(-DEXPOSE_GC)
+
     # Force symbols to be entered in the output file as undefined symbols.
     file(READ "${IOTJS_SOURCE_DIR}/napi/node_symbols.txt" NODE_SYMBOLS)
     string(REGEX REPLACE "[\r|\n]" ";" NODE_SYMBOLS "${NODE_SYMBOLS}")
-    set(NODE_SYMBOLS_LINK_FLAGS "-Wl")
-    # Some tests require the GC to be exposed
-    iotjs_add_compile_flags(-DEXPOSE_GC)
-    foreach(NODE_SYMBOL ${NODE_SYMBOLS})
-      set(NODE_SYMBOLS_LINK_FLAGS
-          "${NODE_SYMBOLS_LINK_FLAGS},-u,${NODE_SYMBOL}")
-    endforeach()
+    if(USING_MSVC)
+      foreach(NODE_SYMBOL ${NODE_SYMBOLS})
+        set(NODE_SYMBOLS_LINK_FLAGS
+            "${NODE_SYMBOLS_LINK_FLAGS} /INCLUDE:_${NODE_SYMBOL}")
+      endforeach()
+    else()
+      set(NODE_SYMBOLS_LINK_FLAGS "-Wl")
+      foreach(NODE_SYMBOL ${NODE_SYMBOLS})
+        set(NODE_SYMBOLS_LINK_FLAGS
+            "${NODE_SYMBOLS_LINK_FLAGS},-u,${NODE_SYMBOL}")
```

Comment:
hmm maybe we could reduce the duplication a bit if the `,-u,` and `/INCLUDE:_` strings are from variables, thus we would end up with something like this:

```
if (MSVC)
  set(LINKER_SYMBOL_INCLUDE "/INCLUDE:_")
else()
  set(LINKER_SYMBOL_INCLUDE ",-u,")
endif()
foreach(NODE_SYMBOL ${NODE_SYMBOLS})
        set(NODE_SYMBOLS_LINK_FLAGS
            "${NODE_SYMBOLS_LINK_FLAGS}${LINKER_SYMBOL_INCLUDE}${NODE_SYMBOL}")
endforeach()
```

Reply:
This won work correctly, because `set(NODE_SYMBOLS_LINK_FLAGS "-Wl")` is also needed for the start of the linker command in case of GCC.